scripting/chat-processor.sp

```diff
@@ -152,23 +151,25 @@ public Action OnSayText2(UserMsg msg_id, BfRead msg, const int[] players, int pl
 		case false: if (BfGetNumBytesLeft(msg)) BfReadString(msg, sMessage, sizeof(sMessage));
 	}
 
-	if (StrEqual(sLastMessage[iSender], sMessage))
-	{
-		return Plugin_Stop;
-	}
-
-	strcopy(sLastMessage[iSender], MAXLENGTH_BUFFER, sMessage);
-
 	if (GetConVarBool(hConVars[4]))
 	{
 		CRemoveColors(sName, sizeof(sName));
 		CRemoveColors(sMessage, sizeof(sMessage));
 	}
 
 	Handle hRecipients = CreateArray();
+	
 	for (int i = 0; i < playersNum; i++)
 	{
-		PushArrayCell(hRecipients, players[i]);
+		if(FindValueInArray(hRecipients, players[i]) == -1) 
+		{
+			PushArrayCell(hRecipients, players[i]);
+		}
+	}
+	
+	if(FindValueInArray(hRecipients, iSender) == -1) 
```

Comment:
I am not entirely sure what you meant here, I took a wild guess in my latest pull :P